internal/workflow/reconciler.go

```diff
@@ -126,7 +126,26 @@ func (r *Reconciler) processNewWorkflow(ctx context.Context, logger logr.Logger,
 
 // templateHardwareData defines the data exposed for a Hardware instance to a Template.
 type templateHardwareData struct {
-	Disks []string
+	Disks      []string
+	Interfaces []netInterface
+}
+
+// netInterface defines a network interface for a Hardware instance.
+// It is different than v1alpha1.Interface so that we don't couple ourselves to the API,
+// especially when we already have a v1alpha2.
+type netInterface struct {
+	MAC    string
+	VLANID string
+	DHCP   dhcp
```

Comment:
I know we've been around this loop but lets go again: do we think DHCP is here to stay with the network interface data? I.e. would we introduce any feature where DHCP isn't specified? If so, how can I easily determine if the DHCP config is available from template language?

Reply:
I think I'm inventing possibilities to challenge this because its meant to be a stable contract. I don't know how productive trying to provide stability here really is though. I feel like we'd be better off just plumbing through the raw `Hardware.Spec` and when that becomes stable this contract inherently becomes stable.

Given we've had numerous requests to get various fields, mostly network interface data, how do you feel about including that and declaring this 'stable' contract a deprecated experiment?

Reply:
@jacobweinstock What did we conclude - I don't recall?